spec/controllers/application_controller_spec.rb

```diff
@@ -1369,4 +1369,45 @@ def hub_path; end
       end
     end
   end
+
+  describe "#ensure_non_admins_reset_passwords_when_forced" do
+    let(:non_admin_user) { create(:team_member_user) }
+    let(:admin_user) { create(:admin_user) }
+
+    context "when a non-admin is logged in" do
+      before do
+        sign_in non_admin_user
+        non_admin_user.update!(forced_password_reset_at: nil)
+      end
+
+      it "redirects to change the password" do
+        get :index
+        expect(response).to redirect_to edit_hub_forced_password_resets_path
+      end
+
+      it "does not redirect if the password was already set" do
+        non_admin_user.update!(forced_password_reset_at: DateTime.now)
+        get :index
+        expect(response).not_to redirect_to edit_hub_forced_password_resets_path
```

Comment:
It's better IMHO to expect the response to `be_ok` if you want to show it's not a redirect. `be_ok` is better than `not_to redirect_to` because presumably you want to show it doesn't redirect at all.

Relatively unimportant.